ClangSharpPInvokeGenerator/Cursors/Exprs/IntegerLiteral.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Diagnostics;
+using ClangSharp;
+
+namespace ClangSharpPInvokeGenerator
+{
+    internal sealed class IntegerLiteral : Expr
+    {
+        private readonly Lazy<string> _rawValue;
+
+        public IntegerLiteral(CXCursor handle, Cursor parent) : base(handle, parent)
+        {
+            Debug.Assert(handle.Kind == CXCursorKind.CXCursor_IntegerLiteral);
+
+            _rawValue = new Lazy<string>(() => {
```

Comment:
We can also relegate cursor specific logic, like extracting the raw text for an `IntegerLiteral` to the type itself, since it isn't really related to the writer.